tf_encrypted/convert/convert.py

```diff
@@ -110,98 +115,138 @@ class InvalidArgumentError(Exception):
     pass
 
 
-def find_leaves(special_ops, graph_def, lookahead):
-    potential_leaf_ops = []
-    graph_plusminus_special = []
-
+def find_leaves(scope, subscope_map):
+    """
+    Assembles input and output leaf nodes of the subgraph represented by subscope_map.
+    """
+
+    input_leaves = []
+    output_leaves = list(subscope_map.keys())
+    for name, node in subscope_map.items():
+        for input in node.input:
+            if match_numbered_scope(scope, input) is None:
+                input_leaves.append(input)
+            if input in output_leaves:
+                output_leaves.remove(input)
+    seen = set()
+    adder = seen.add
+    input_leaves = [x for x in input_leaves if not (x in seen or adder(x))]
+
+    return input_leaves, output_leaves
+
+
+def get_intermediaries(specop_scope, subscope_map):
```

Comment:
Can you explain what you mean by an intermediary here in a comment? Its not immediately obvious to me.

Reply:
`An intermediary op for a subgraph is an op that is neither an input nor an output to that subgraph.` would that work?

Reply:
I'd call that on internal op

Reply:
Yep sounds good!